addons/website_sale/models/res_config_settings.py

```diff
@@ -110,6 +117,24 @@ def _inverse_account_on_checkout(self):
             else:
                 record.website_id.auth_signup_uninvited = 'b2b'
 
+    # === ONCHANGE METHODS === #
+
+    @api.onchange('selectable_pricelist_ids')
+    def onchange_selectable_pricelist_ids(self):
+        for res_config in self:
+            ProductPricelist = self.env['product.pricelist']
+            website_id = res_config.website_id
+            selected_pricelists = ProductPricelist.search([
+                ('id', 'in', res_config.selectable_pricelist_ids.ids),
+            ])
+            for pricelist in selected_pricelists:
+                pricelist.website_ids |= website_id
+            unselected_pricelists = ProductPricelist.search(
+                [('website_ids', 'any', [('id', '=', website_id.id)])]
+            ) - selected_pricelists
+            for pricelist in unselected_pricelists:
+                pricelist.website_ids -= website_id
```

Comment:
Never change another model in an onchange !
It means that even if you don't save your modified settings, it'd still have had an impact on existing records already.